### ISSUE TITLE
New S3 bucket ownership default breaks the S3 streamwrapper mkdir method.

### ISSUE THREAD

**Issue by tbenice** [aws/aws-sdk-php#2674]
### Describe the bug

April 2023 S3 started defaulting all new buckets to 'Bucket owner enforced' ownership setting (https://aws.amazon.com/about-aws/whats-new/2022/12/amazon-s3-automatically-enable-block-public-access-disable-access-control-lists-buckets-april-2023/) . Because of this, any createbucket call with acl settings returns a 400 error "InvalidBucketAclWithObjectOwnership". 

This would be fine except that the streamwrapper class mkdir method uses acls by default, and so any use of mkdir returns the 400 error.

### Expected Behavior

Using mkdir method of the streamwrapper class should create a bucket successfully.

### Current Behavior

mkdir method always returns a 'InvalidBucketAclWithObjectOwnership' error.

### Reproduction Steps

mkdir("s3://my-bucket-name");

### Possible Solution

in the mkdir method try creating the bucket using 'DeleteBucketOwnershipControls' to allow acls, or remove any acl setting from the method.

### SDK version used

^3.0

### Environment details (Version of PHP (`php -v`)? OS name and version, etc.)

php 8, macos, Pantheon hosting environment.

**Comment by stobrien89** [comment]
Hi @tbenice,

Sorry to hear about the issues.  I imagine we can just unset the `ACL` option for the `CreateBucket` method, but how we handle `createSubfolder` will require a little more thought.  Ideally we'd want to find out when the bucket was created and proceed from there, but for that we'll need to contact S3.  I'll try to have a fix out for this soon.

**Comment by stof** [comment]
Given that the configuration of the bucket can still be changed, `createSubfolder` should probably choose its behavior based on the metadata of the bucket rather than choosing it based on the default value of the bucket creation at the time of creation of the bucket.

**Comment by stobrien89** [comment]
Apologies for the delay here.  We decided to leave `createSubfolder` unaltered for backward compatibility reasons.  Some users with buckets created prior to April 2023 might expect an ACL to be automatically determined and applied on their behalf.  We considered the metadata route, but didn't want to make these abstractions any more expensive by adding additional S3 calls.  We added a caveat to our `mkDir()` documentation [here](https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/s3-stream-wrapper.html#working-with-buckets).  

The new S3 access control paradigms are too fine-grained for us to reasonably translate into permissions levels, so it's unlikely there will be any additional work in that regard.
